services/galley/src/Galley/API/MLS/Message.hs

```diff
@@ -499,6 +507,28 @@ postMLSMessageToLocalConv qusr senderClient con smsg lcnv = case rmValue smsg of
     propagateMessage qusr lconv cm con (rmRaw smsg)
 
     pure events
+  where
+    gtcToConv :: GlobalTeamConversation -> Data.Conversation
```

Comment:
The `undefined` here is not good, just a horrible runtime bug waiting to happen. Also, this function could be extracted and reused, since it (or a version of it) is defined at least twice. 

If you decide not to go with the sum type solution for the `Conversation` type, then I think the only way would be to make the creator field optional, and adapt the code everywhere else.